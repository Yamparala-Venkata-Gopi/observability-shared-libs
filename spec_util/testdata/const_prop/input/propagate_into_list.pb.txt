# MethodTemplate proto

method_id: {
  name: "new_method_with_refs"
}

arg_templates: {
  key: "arg1"
  value: {
    list_template: {
      elem_templates: {
        # This elem should become a constant because it refers to a constant.
        ref: {
          method_index: 0
          arg_ref: {
            key: "primitive_constant_arg"
            data_ref: {
              primitive_ref: {}
            }
          }
        }
      }
      elem_templates: {
        ref: {
          method_index: 0
          response_ref: {
            key: "response_1"
            data_ref: {
              primitive_ref: {}
            }
          }
        }
      }
    }
  }
}
