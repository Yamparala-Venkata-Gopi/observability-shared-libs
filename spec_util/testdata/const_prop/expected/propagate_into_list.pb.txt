# MethodTemplate proto

method_id: {
  name: "new_method_with_refs"
}

arg_templates: {
  key: "arg1"
  value: {
    list_template: {
      elem_templates: {
        value: {
          primitive: {
            type_hint: "string_constant_1"
            string_value: {
              value: "7c73565e-0d62-49b0-84e7-3063f4d53d9c"
            }
          }
        }
      }
      elem_templates: {
        ref: {
          method_index: 0
          response_ref: {
            key: "response_1"
            data_ref: {
              primitive_ref: {}
            }
          }
        }
      }
    }
  }
}
